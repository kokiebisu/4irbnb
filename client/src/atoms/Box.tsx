import React from 'react';
import styled from 'styled-components';

interface ElementProps {
  children?: React.ReactNode;
  styles?: any;
}

interface StyledProps {
  styles?: any;
}

const Element = styled.div<StyledProps>`
  ${({ styles }) => styles}
`;

export default ({ children, ...props }: ElementProps) => {
  return <Element {...props}>{children}</Element>;
};
