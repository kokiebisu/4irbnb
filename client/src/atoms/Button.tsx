import React from 'react';
import styled, { css } from 'styled-components';
import { motion } from 'framer-motion';

export interface ElementProps extends StyledProps {
  children?: React.ReactNode;
  onPress: () => void;
  className?: string;
}

interface StyledProps {
  styles?: any;
  size?: 'small' | 'medium' | 'large';
  inverse?: boolean;
}

const Element = styled(motion.button)<StyledProps>`
  border-radius: 8px;
  font-size: 18px;
  border: none;
  ${({ styles }) => styles};
  ${({ size }) =>
    size === 'small' &&
    css`
      padding: 15px 22px;
    `};
  ${({ size }) =>
    size === 'medium' &&
    css`
      padding: 15px 50px;
    `};
  ${({ size }) =>
    size === 'large' &&
    css`
      padding: 15px 100px;
    `};
  ${({ inverse }) =>
    inverse &&
    css`
      border: 1px solid black;
      background-color: transparent;
    `};
  &:hover {
    cursor: pointer;
  }
  &:focus {
    outline: none;
  }
`;

export default ({
  size = 'medium',
  children,
  onPress = () => console.log('hello'),
  ...props
}: ElementProps) => {
  return (
    <Element size={size} {...props} onClick={onPress}>
      {children}
    </Element>
  );
};

export type CustomProps = {
  type?: string;
  onPress: () => void;
  size?: 'small' | 'medium' | 'large';
  inverse?: boolean;
  name?: string;
};
