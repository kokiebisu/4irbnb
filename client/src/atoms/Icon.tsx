import React from 'react';
import styled from 'styled-components';

// svgs
import { Search, Globe, BottomArrow } from 'assets/svg';

interface ElementProps {
  name: string;
  styles?: any;
}

interface StyledProps {
  styles?: any;
}

interface MapProps {
  [key: string]: () => JSX.Element;
}

const items: MapProps = {
  Search: Search,
  Globe: Globe,
  BottomArrow: BottomArrow,
};

const Element = styled.div<StyledProps>`
  ${({ styles }) => styles}
`;

export default ({ name, ...props }: ElementProps) => {
  const Component: any = items[name];

  return (
    <Element {...props}>
      <Component />
    </Element>
  );
};
