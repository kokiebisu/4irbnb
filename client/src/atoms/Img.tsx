import React from 'react';
import styled from 'styled-components';

interface ElementProps {
  src: any;
  alt: string;
  className?: string;
}

const Element = styled.img`
  max-width: 100%;
  height: auto;
`;

export interface Props extends ElementProps {}

export const Img: React.FC<Props> = ({ ...props }) => {
  return <Element {...props} />;
};
