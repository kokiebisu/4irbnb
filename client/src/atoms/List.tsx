import React from 'react';
import styled from 'styled-components';

interface ElementProps {
  className?: string;
}

interface StyledProps {
  styles?: any;
}

const Element = styled.li<StyledProps>`
  ${({ styles }) => styles}
`;

export interface ListProps extends ElementProps, StyledProps {}

export const List: React.FC<ListProps> = ({ ...props }) => {
  return <Element {...props} />;
};
