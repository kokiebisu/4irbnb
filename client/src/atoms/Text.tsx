import React from 'react';
import styled from 'styled-components';
import { motion } from 'framer-motion';

interface ElementProps {
  children?: React.ReactNode;
  className?: string;
}

interface StyledProps {
  styles?: any;
}

const Element = styled(motion.p)<StyledProps>`
  margin: 0;
  font-family: 'Airbnb-Cereal';
`;

interface Props extends ElementProps, StyledProps {}

export const Text: React.FC<Props> = ({ children, ...props }) => {
  return <Element {...props}>{children}</Element>;
};
