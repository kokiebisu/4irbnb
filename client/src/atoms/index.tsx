export * from 'atoms/Box';
export * from 'atoms/Button';
export * from 'atoms/Icon';
export * from 'atoms/Img';
export * from 'atoms/Text';
export * from 'atoms/List';
