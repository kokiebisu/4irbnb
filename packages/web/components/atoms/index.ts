export * from './bullet';
export * from './button';
export * from './icon';
export * from './input';
export * from './video';
