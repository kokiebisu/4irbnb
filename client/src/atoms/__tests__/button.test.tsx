import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Button } from 'atoms';

describe('Base Button', () => {
  it('renders correctly', () => {
    const testingData = {
      children: 'test',
      action: () => console.log('test'),
    };

    const tree = renderer
      .create(<Button sm onPress={testingData.action} {...testingData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
