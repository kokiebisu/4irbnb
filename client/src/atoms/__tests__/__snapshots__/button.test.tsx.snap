// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Button renders correctly 1`] = `
.c0 {
  border-radius: 8px;
  font-size: 18px;
  border: none;
  padding: 15px 22px;
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

<button
  action={[Function]}
  className="c0"
  onClick={[Function]}
>
  test
</button>
`;
